src/libxrpl/protocol/STIssue.cpp

```diff
@@ -65,18 +89,41 @@ STIssue::getSType() const
     return STI_ISSUE;
 }
 
+std::string
+STIssue::getText() const
+{
+    return asset_.getText();
+}
+
 Json::Value
 STIssue::getJson(JsonOptions) const
 {
-    return to_json(issue_);
+    Json::Value jv;
+    asset_.setJson(jv);
+    return jv;
 }
 
 void
 STIssue::add(Serializer& s) const
 {
-    s.addBitString(issue_.currency);
-    if (!isXRP(issue_.currency))
-        s.addBitString(issue_.account);
+    if (holds<Issue>())
+    {
+        s.addBitString(asset_.get<Issue>().currency);
+        if (!isXRP(asset_.get<Issue>().currency))
+            s.addBitString(asset_.get<Issue>().account);
+        auto const& issue = asset_.get<Issue>();
+        s.addBitString(issue.currency);
+        if (!isXRP(issue.currency))
+            s.addBitString(issue.account);
+    }
+    else
+    {
+        auto const& issue = asset_.get<MPTIssue>();
+        s.addBitString(issue.getIssuer());
+        s.addBitString(noAccount());
+        std::uint32_t sequence;
+        memcpy(&sequence, issue.getMptID().data(), sizeof(sequence));
```

Comment:
... here (lost `s.add32`)

Reply:
I guess I should not have used vscode for suggestions :-/

Reply:
I usually copy and paste suggestions into my IDE because suggestions often times are not formatted...

Reply:
well, that did not work this time.

Reply:
Pushed a fix.